### PR TITLE
(PUP-9458) Download CRL explicitly

### PR DESCRIPTION
**Pull request by joshcooper**
Restore the ability to download a missing CRL when the agent already has a private key and cert, and the ability for the agent to use an incomplete cert chain, and have the server resolve it based on CA certs the server has locally.

### CODE REVIEW

**Comment by joshcooper** [comment]
jenkins please test this with servertests

File: spec/unit/ssl/host_spec.rb [hunk 1]
```diff
@@ -308,53 +309,60 @@ def base_json_comparison(result, json_hash)
...
+      stub_request(:get, %r{puppet-ca/v1/certificate/#{@host.name}}).to_return(status: 404)
+
       @host.certificate
       actual_ca_bundle = Puppet::FileSystem.read(Puppet[:localcacert])
       expect(actual_ca_bundle).to match(/BEGIN CERTIFICATE.*END CERTIFICATE.*BEGIN CERTIFICATE/m)
     end
 
-    it "should return nil if it cannot find a CA certificate" do
-      @host.expects(:ensure_ca_certificate).returns(false)
+    it "should raises cannot find a CA certificate" do
```
Comment (kris-bosland) [inline spec/unit/ssl/host_spec.rb]: Maybe "should raise if it cannot find a CA certificate" ?

Comment (joshcooper) [reply spec/unit/ssl/host_spec.rb]: Done

**Comment by joshcooper** [comment]
jenkins please test this with servertests

**Comment by joshcooper** [comment]
Passed puppetserver CI using 

```diff
diff --git a/acceptance/config/beaker/options.rb b/acceptance/config/beaker/options.rb
index d7bb5c72..c4fbebd3 100644
--- a/acceptance/config/beaker/options.rb
+++ b/acceptance/config/beaker/options.rb
@@ -11,4 +11,4 @@
  "puppetserver-confdir"=>"/etc/puppetlabs/puppetserver/conf.d",
  "puppetserver-config"=>
   "/etc/puppetlabs/puppetserver/conf.d/puppetserver.conf",
- :puppet_build_version=>"e625403d9d9ef287f7242840fbfbf54cac7130cc"}
+ :puppet_build_version=>"0ceffe880a858263cec007cf5a84d8e42f564a4e"}
```

And running:
```
GENCONFIG_LAYOUT=redhat7-64ma-redhat7-64a PACKAGE_BUILD_VERSION=6.2.2.SNAPSHOT.2019.03.09T1350 BEAKER_TESTSUITE=acceptance/suites/tests ./acceptance/scripts/generic/testrun.sh -p
...
n3ofm3qe0795tfy.delivery.puppetlabs.net (redhat7-64-1) 00:46:18$ puppet agent --confdir /tmp/jetty_external_root_ca.QQWmDD/etc/agent --vardir /tmp/jetty_external_root_ca.QQWmDD/etc/agent/var --test
  Info: Certificate revocation is disabled, skipping CRL download
  Warning: The issuer '/CN=Intermediate CA (agent-ca)/emailAddress=test@example.org/O=Example Org, LLC/OU=Server Operations' of certificate '/CN=agent1.example.org' cannot be found locally
  Info: Using configured environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
...
              - Test Case Summary for suite 'tests' -
       Total Suite Time: 1441.70 seconds
      Average Test Time: 90.11 seconds
              Attempted: 16
                 Passed: 16
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 16
```

**Pull request merged**